blueocean-core-js/src/js/components/RunButton.jsx

```diff
@@ -86,14 +84,18 @@ export class RunButton extends Component {
         showRunButton = showRunButton && permit(this.props.runnable).start();
         showStopButton = showStopButton && permit(this.props.runnable).stop();
 
-        const runLabel = this.props.runText || translate('toast.run', {
-            defaultValue: 'Run',
-        });
-        let stopLabel = this.state.stopping ? translate('toast.stopping', {
-            defaultValue: 'Stopping ...',
-        }) : translate('toast.stop', {
-            defaultValue: 'Stop',
-        });
+        const runLabel =
+            this.props.runText ||
+            translate('toast.run', {
+                defaultValue: 'Run',
+            });
+        let stopLabel = this.state.stopping
+            ? translate('toast.stopping', {
```

Comment:
oh how i hate `translate()`

Reply:
Yes.

Reply:
We really ought to be able to remove most of the `{ defaultValue: 'my label' }` stuff since the `i18n` is supposed to load first. Maybe not the case when there isn't an active extension point on the screen? We do need to get that fixed and then kill `defaultValue` with fire, it really hurts readability IMO.